### DISCUSSION TITLE
Summarizing court opinions

### DISCUSSION THREAD

**Discussion by legaltextai** [freelawproject/courtlistener#3476] [General]
https://helpmefindcase.com/ 
This is an early prototype. 
If you are a practicing lawyer, what functionality would you like to see added? 
This is my pet project. No plans to charge for it. At least for as long as I can afford the database bill. 
Cheers!

**Comment by mlissner** [comment]
Pretty cool! I tried a tough question, and it got me closer to an answer: 

> at&t built a cell phone tower on my lot and has stopped paying their lease for it. Can I tear it down and throw it away?

Really neat to see how quickly these things can answer or at least advance tough questions like this. 

What are your DB stats like?

**Comment by legaltextai** [comment]
That one would fall under a state issue, so the answer may vary state by state. You tried CA probably?  
The database is at ~500GB now. Broken into tables for states and federal. 
I am pretty sure we'll see more and more projects like this popping up in the nearest future. Rich legal texts + AI are destined for each other. 
This would not be possible without your and your contributors' efforts to make this data publicly available. Thank you and Viva le CL!

**Reply by mlissner** [reply]
I was doing federal, because I'm lazy, but you're right. I tried in CA, and got a mish-mash of results. The cases selected didn't seem to have much to do with my question, but the summary did. Interesting. 

![image](https://github.com/freelawproject/courtlistener/assets/236970/0502f841-804c-48b9-8169-26db328056e5)

That's sort of interesting because it leaves me wishing I knew which opinions supported the summary, since the cited ones don't seem to. Hm.

One bit of feedback, if you can make it type out the answers more quickly, that might be nice. I think I can skim faster than it is typing right now.

**Comment by legaltextai** [comment]
When providing an answer, the model analyzes all relevant excerpts. I need to adjust the prompt so that it summarizes all relevant cases, not just 2-3. 
I plan to separate the two functionalities: 1) finding relevant cases based on facts and 2) summarizing them and providing an answer to the query. Each relevant case will have IRAC summary. 
The streaming comes directly from the model, Anthropic in this case. There is nothing I can do on my side to speed it up unfortunately.

**Reply by tonystark262** [reply]
@legaltextai I have also built out a similar system to search for documents. There was no alternative way to communicate with you so I am writing here. I am interested to know more about what you are doing and if we can collaborate on this. Let me know if you want to connect.

**Reply by legaltextai** [reply]
sure @tonystark262 . always happy to connect with like minded humans! admin at legaltext.ai . also am interested to check what you have built.

**Reply by tonystark262** [reply]
Thanks, sent you an email!

**Comment by legaltextai** [comment]
@mlissner since you asked for answers to be typed faster, i've made some changes. check it out. but with something simpler than at&t towers :-) hope you like it [helpmefindcase.com](https://helpmefindcase.com)
